docs/5_mobile_native_app_generation.md

```diff
@@ -543,8 +548,8 @@ mobile-native/
 │   │   └── run/       # Deployment, validation, live feedback
 │   ├── schemas/   # Zod schemas for tool inputs/outputs
 │   └── utils/     # Shared utilities and CLI integrations
-├── resources/     # Template metadata and documentation
-│   ├── templates/     # Mobile SDK template descriptions and extension guides
+├── resources/     # Template repository and documentation
+│   ├── SalesforceMobileSDK-Templates/  # Git submodule: Official Salesforce Mobile SDK Templates
```

Comment:
Wouldn't using submodule create a version specific coupling? Do we not want to use `forceios`/`forcedroid` to "fill" the `templates` folder(as you had before) with `listTemplate` as @jhorst11 updated(and in the future also complement that list of templates with metadata when the `sfmobile-native-template-discovery` is finished)?